fecgov/fec-cms
# Feature/4351 debts section

## Summary

- Resolves #4351  
Adding the debts section!

## Reviewers
- Paul for purpose and expertise (general and detailed)
- UX for, uh, UX concerns
- Back-end to verify proper integration
- Front-end for functionality

## Impacted areas of the application

- Updated `/data/browse-data/?tab=loans-debts` to activate the Debts section
- Created `/data/debts/`
- Changed range filters' naming logic, which should only add flexibility

## Screenshots

Current
![image](https://user-images.githubusercontent.com/26720877/106168770-ab138200-615c-11eb-8c88-7194d49c5b85.png)

New
![image](https://user-images.githubusercontent.com/26720877/106168871-c3839c80-615c-11eb-92b0-44311d8fac2f.png)
![image](https://user-images.githubusercontent.com/26720877/106168923-cda59b00-615c-11eb-9a6e-b9acc390fe0f.png)

Details panel
![image](https://user-images.githubusercontent.com/26720877/106168964-d9915d00-615c-11eb-8ed3-b86cb17451d6.png)



## Related PRs

None

## How to test

- Pull branch
- `npm i`, `npm run build`
- `./manage.py runserver`
- Go to the [loans/debts section](http://127.0.0.1:8000/data/browse-data/?tab=loans-debts)
- Play with the debts [/data/debts/](http://127.0.0.1:8000/data/debts/)
- **NOTE**: the dollar filters only work on Dev right now and could change if someone pushes a change before the [corresponding API ticket](https://github.com/fecgov/openFEC/issues/4755) is merged
- Because of the change to the range filters' names:
  - Check other datatables' range filters where there is one 'amount' range (not a date range)
  - Check where there are multiple 'amount' ranges on the same table. Looking at the code, these have two or more range.amount filters:
    - debts (obviously)
    - loans
    - reports
  - Range filters should function like normal: filter data properly, add & remove chiclets properly


____
